### PR TITLE
Fix signal feedback on IR

### PR DESCRIPTION
**Pull request by Xyene**

### CODE REVIEW

File: dmoj/cptbox/_cptbox.pyx [hunk 1]
```diff
@@ -83,7 +83,6 @@ cdef extern from 'ptbox.h' nogil:
 
     cdef int PTBOX_EXIT_NORMAL
     cdef int PTBOX_EXIT_PROTECTION
-    cdef int PTBOX_EXIT_SEGFAULT
```
Comment (Riolku) [inline dmoj/cptbox/_cptbox.pyx:86]: Why are we deleting this right now?

Otherwise code looks fine and explanation makes sense.

Comment (Xyene) [reply dmoj/cptbox/_cptbox.pyx:86]: https://github.com/DMOJ/judge-server/pull/1061/commits/b62a61496e050ff74ee8d90d95905fd4a3a7a084

Comment (Riolku) [reply dmoj/cptbox/_cptbox.pyx:86]: Sure, makes sense. Honestly would've expected this in a separate PR.

**Review by Riolku — APPROVED** [review]
LGTM

File: dmoj/cptbox/_cptbox.pyx [hunk 2]
```diff
@@ -588,7 +587,25 @@ cdef class Process:
...
+        # thread sends `SIGVTALRM`), in which case a successful execution would
+        # have a nonzero `_signal`. This would lead the judge into displaying
+        # confusing output (e.g., "IR (virtual timer expired)").
+        #
+        # Avoid such cases. While we can't for sure distinguish signals that
+        # killed a process from signals that were part of normal execution, a
+        # zero or positive exit code is a good proxy.
+        #
+        # A more aggressive check here would be to ensure `self._exitcode ==
+        # -self.signal`.
```
Comment (quantum5) [inline dmoj/cptbox/_cptbox.pyx:604]: What do you mean you can't know for sure? Non-negative exit codes imply process didn't exit due to a signal, and this is guaranteed by `cptbox`: https://github.com/DMOJ/judge-server/blob/dbd148900ac7d052b5afd751ab0413cec72295a8/dmoj/cptbox/ptproc.cpp#L352

**Review by quantum5 — APPROVED** [review]

**Pull request merged**